src/Uno.UI/UI/Xaml/Media/AcrylicBrush/Android/RealtimeBlurView.Android.cs

```diff
@@ -0,0 +1,413 @@
+ limitations under the License.
+
+------------------------------------------------------------------------------
+ Adapted to csharp by Jean-Marie Alfonsi
+------------------------------------------------------------------------------
+*/
+
+using System;
+using System.Diagnostics;
+using Android.App;
+using Android.Content;
+using Android.Graphics;
+using Android.Runtime;
+using Android.Util;
+using Android.Views;
+
+namespace Uno.UI.Xaml.Media
+{
+	internal partial class RealtimeBlurView : View
+	{
+		private float mDownsampleFactor = 4;
+		private int mOverlayColor; // default #aaffffff
+		private float mBlurRadius = 10; // (0 < r <= 25)
+
+		private IBlurImpl mBlurImpl;
+		private bool mDirty;
+		private Bitmap mBitmapToBlur, mBlurredBitmap;
+		private Canvas mBlurringCanvas;
+		private bool mIsRendering;
+		private Paint mPaint;
+		private readonly Rect mRectSrc = new Rect();
+		private readonly Rect mRectDst = new Rect();
+		// mDecorView should be the root view of the activity (even if you are on a different window like a dialog)
+		private View mDecorView;
+		// If the view is on different root view (usually means we are on a PopupWindow),
+		// we need to manually call invalidate() in onPreDraw(), otherwise we will not be able to see the changes
+		private bool mDifferentRoot;
+		private static int RENDERING_COUNT;
+
+		private readonly PreDrawListener _preDrawListener;
+
+		public RealtimeBlurView(Context context) : base(context)
+		{
+			_preDrawListener = new PreDrawListener(this);
+			mBlurImpl = GetBlurImplementation(); // provide your own by override getBlurImpl()			
+			mBlurRadius = TypedValue.ApplyDimension(ComplexUnitType.Dip, 10, context.Resources.DisplayMetrics);
+			//mDownsampleFactor = a.getFloat(R.styleable.RealtimeBlurView_realtimeDownsampleFactor, 4);
+
+			mPaint = new Paint();
+		}
+
+		public void SetOverlayColor(int color, bool invalidate = true)
+		{
+			if (mOverlayColor != color)
+			{
+				mOverlayColor = color;
+				if (invalidate)
+				{
+					Invalidate();
+				}
+			}
+		}
+
+		protected IBlurImpl GetBlurImplementation() => new AndroidStockBlur();
+
+		private class PreDrawListener : Java.Lang.Object, ViewTreeObserver.IOnPreDrawListener
+		{
+			private readonly JniWeakReference<RealtimeBlurView> _weakBlurView;
+
+			public PreDrawListener(RealtimeBlurView blurView)
+			{
+				_weakBlurView = new JniWeakReference<RealtimeBlurView>(blurView);
+			}
+
+			public PreDrawListener(IntPtr handle, JniHandleOwnership transfer)
+				: base(handle, transfer)
+			{
+			}
+
+			public bool OnPreDraw()
+			{
+				if (!_weakBlurView.TryGetTarget(out var blurView))
+				{
+					return false;
+				}
+
+				int[] locations = new int[2];
+				Bitmap oldBmp = blurView.mBlurredBitmap;
+				View decor = blurView.mDecorView;
+				if (decor != null && blurView.IsShown && blurView.Prepare())
+				{
+					bool redrawBitmap = blurView.mBlurredBitmap != oldBmp;
+					oldBmp = null;
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'oldBmp'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5932087)